### PR TITLE
[#9595] Migrate GET /session/students/response to RESTFul API

### PR DESCRIPTION
**Pull request by crizyli**
Part of #9595

Source code from @xpdavid : https://github.com/xpdavid/teammates/tree/try-respondent-optimization

### CODE REVIEW

File: src/web/app/components/sessions-table/student-list-info-table/student-list-info-table.component.ts [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+  studentListInfoTableSortBy: SortBy = SortBy.NONE;
+  studentListInfoTableSortOrder: SortOrder = SortOrder.ASC;
+
+  constructor() { }
+
+  ngOnInit(): void {
+  }
+
+  /**
+   * Sort the students in according to selection option.
```
Comment (xpdavid) [inline src/web/app/components/sessions-table/student-list-info-table/student-list-info-table.component.ts]:
```suggestion
   * Sorts the students according to selection option.
```

File: src/web/app/components/sessions-table/student-list-info-table/student-list-info-table.component.ts [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        this.studentListInfoTableSortOrder === SortOrder.DESC ? SortOrder.ASC : SortOrder.DESC;
+
+    this.studentListInfoTableRowModelsChange.emit(
+        this.studentListInfoTableRowModels.map((oldModel: StudentListInfoTableRowModel) => Object.assign({}, oldModel))
+            .sort(this.sortRowsBy(by, this.studentListInfoTableSortOrder)),
+    );
+  }
+
+  /**
+   * Handler a when a row of the table is selected.
```
Comment (xpdavid) [inline src/web/app/components/sessions-table/student-list-info-table/student-list-info-table.component.ts]:
```suggestion
   * Handle the event when a row of the table is selected.
```

File: src/web/app/pages-instructor/instructor-session-modal-page.component.ts [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+export abstract class InstructorSessionModalPageComponent extends InstructorSessionBasePageComponent {
+
+  protected constructor(router: Router,
+                        httpRequestService: HttpRequestService,
+                        statusMessageService: StatusMessageService,
+                        navigationService: NavigationService,
+                        feedbackSessionsService: FeedbackSessionsService,
+                        feedbackQuestionsService: FeedbackQuestionsService,
+                        protected modalService: NgbModal,
+                        protected studentService: StudentService) {
```
Comment (damithc) [inline src/web/app/pages-instructor/instructor-session-modal-page.component.ts]: Shouldn't the indentation be twice the normal?

Comment (xpdavid) [reply src/web/app/pages-instructor/instructor-session-modal-page.component.ts]: They are automatically aligned by IntelliJ. Though it "violates" the coding standard we published, I am OK with the indentation for the constructor as it is more clearly showing the services used.

**Pull request merged**